CARDS-1808: Add the ability to search questionnaire descriptions in the new form dialog

marta- opened the pull request:
[Jira link](https://phenotips.atlassian.net/browse/CARDS-1808)

marta- committed:
CARDS-1808: Add the ability to search questionnaire descriptions in the new form dialog

sashaandjic reviewed (approved):
tested
![image](https://user-images.githubusercontent.com/23243244/174184772-9f988e93-56f8-45ea-9eb1-b0c1ed98f643.png)
![image](https://user-images.githubusercontent.com/23243244/174185027-b85e8fec-dcd9-4952-b1be-0b9c23d81503.png)

sdumitriu commented on modules/data-entry/src/main/frontend/src/dataHomepage/NewFormDialog.jsx:
```diff
@@ -298,7 +298,7 @@ function NewFormDialog(props) {
                   let conditions = [];
                   if (query.search) {
-                    conditions.push(`CONTAINS(n.'title', '*${query.search}*')`);
+                    conditions.push(`CONTAINS(n.'title', '*${query.search}*') or CONTAINS(n.'description', '*${query.search}*')`);
```
This needs to be wrapped in `()`.

marta- replied:
Done in 15b713f.

marta- committed:
CARDS-1808: Add the ability to search questionnaire descriptions in the new form dialog

Addressing review comments: wrap `OR`-ed conditions in brackets to avoid interference with other conditions that might appear in the query
